core/src/main/scala/com/banno/kafka/consumer/ConsumerOps.scala

```diff
@@ -353,4 +353,101 @@ case class ConsumerOps[F[_], K, V](consumer: ConsumerApi[F, K, V]) {
       .recordStream(pollTimeout)
       .evalMap(r => process(r) <* consumer.commitSync(r.nextOffset))
 
+  /** Returns a stream that processes records using the specified function,
+    * committing offsets for successfully processed records, either after
+    * processing the specified number of records, or after the specified time
+    * has elapsed since the last offset commit. If the processing function
+    * returns a failure, offsets for records successfully processed before that
+    * failure will be committed, and then the stream will halt with that
+    * failure. This is at-least-once processing: after a restart, the record
+    * that failed will be reprocessed. In some use cases this pattern is more
+    * appropriate than just using auto-offset-commits, since it will not commit
+    * offsets for failed records when the consumer is closed. The consumer must
+    * be configured to disable offset auto-commits.
+    */
+  def processingAndCommitting[A](
+      pollTimeout: FiniteDuration,
+      maxRecordCount: Long = 1000L,
+      maxElapsedTime: FiniteDuration = 60.seconds,
+  )(
+      process: ConsumerRecord[K, V] => F[A]
+  )(implicit C: Clock[F], S: Concurrent[F]): Stream[F, A] =
+    for {
+      state <- Stream.eval(
+        C.realTime
```

Comment:
I think this should be monotonic time?  Also L392.

Reply:
I'm never 100% clear which `Clock` method to use, but this is trying to get the current system time, so `realTime` seems like the right one to use?

Reply:
Are you getting the time to record a timestamp in a distributed system?  That's real time.

Are you getting the time to later calculate an elapsed time in the same system?  That's monotonic time.

Reply:
Distributed system, no. e.g. not being used as a timestamp field written to postgres/kafka. The timestamps are indeed only being used to calculate a duration in the same system. So you're right it should be `monotonic`. Fixed in https://github.com/Banno/kafka4s/pull/782/commits/a1d907bc6566e1f49baa988abbd0f6e106d88da2.